### ISSUE TITLE
Changes in 3103

### ISSUE THREAD

**Issue by yeefom** [braver/Solarized#51]
Hi @braver!

Thank you for the better solarized theme! Solarized Light is by far my favorite color scheme. As there are some major changes regarding scoping for languages such as JavaScript (especially JavaScript!) in the new Sublime update, I wonder if you plan to update it to catch up with the new update!

Thanks in advance!

**Comment by braver** [comment]
@yeefom I'm definitely keeping an eye on it. It's very much in flux still though, 3104 (dev release) apparently brings a lot of fixes for JS. Of course the new beta is a big thing so lots of people run into things. I'm kinda thinking they'll do a new beta shortly to address issues people are having and that would be my key to tweak Solarized to match. Otherwise there's a big chance I have to backtrack on the changes soon.

**Comment by yeefom** [comment]
Thank you @braver! Really looking forward to the update!

**Comment by yeefom** [comment]
Hey braver, I’m playing around with ES6 recently and came across [Babel](https://github.com/babel/babel-sublime). It seems that it does pretty perfect syntax highlighting for ES5, ES6 and React/JSX. Wonder if you would consider optimize your awesome theme for Babel?

**Comment by braver** [comment]
Is there anything specific you're looking for that's now missing when highlighting Babel? I have zero experience with Babel, so I won't go out and try to optimise it myself, but specific requests to highlight something etc. I don't mind looking into.

**Comment by yeefom** [comment]
Hi @braver, thanks! Please take a look when you have time.

As showing in the image, everything in front of a dot is in blue, parameters, and methods are all in blue. I guess ideally function name/variable name and methods would have the same color. Also not everything in front of a dot needs a special color.

![](https://cldup.com/yI3qjqVbUf.png)

Here’s a theme that’s optimized for babel (Oceanic Next) as a reference.

![](https://cldup.com/fBK30-8Ktq.png)

Babel also probably has a special notation for keywords such as `this` and `super`. Here is how they are in Solarized.

![](https://cldup.com/GQ3OVXASSS.png) 

In comparison:

![](https://cldup.com/eP93tr3Mix.png)

Thanks again!

**Comment by braver** [comment]
Oh, that makes sense! How about this:

before
<img width="250" alt="before" src="https://cloud.githubusercontent.com/assets/2543659/14185063/82b497d4-f778-11e5-8492-1dcda9f58bb1.png">

after
<img width="255" alt="after" src="https://cloud.githubusercontent.com/assets/2543659/14185068/864d2c76-f778-11e5-901a-e45ed26aaa3a.png">

I think it looks a bunch better, I dig the highlighting for `this` even in normal JS. People will probably lynch me for changing *anything* about this :sunglasses: @yeefom What do you say? Anything else?

**Comment by yeefom** [comment]
That looks beautiful! Thanks a bunch! I couldn’t see in your screenshots, but how about the color of function name and method names like in the first two screenshots. For example for `var avg = function() {}` and `rawArgs.forEach()`, `avg` and `forEach` would have the same color (orange-ish yellow), and `rawArgs` wouldn’t have a special color (should be grey).

**Comment by braver** [comment]
`rawArgs` would be grey, but `avg` is yellow (for 'entity.name.function') and `forEach` (for 'variable'). I'm adding a special case for `variable.function.js`, to change that:

<img width="283" alt="screen shot 2016-03-31 at 20 17 01" src="https://cloud.githubusercontent.com/assets/2543659/14186224/9f9b6fc6-f77d-11e5-94fe-996344bb05f7.png">

**Comment by braver** [comment]
Could you perhaps test this for me? If you open Solarized (light).thTheme, all the way at the bottom there is a single block for JavaScriptNext, you can replace that with this:

```xml
<!-- JavaScript and Babel/ES2015 -->
		<dict>
			<key>name</key>
			<string>JavaScript Variables</string>
			<key>scope</key>
			<string>variable.other.readwrite.js, variable.other.object.js</string>
			<key>settings</key>
			<dict>
				<key>foreground</key>
				<string>#657b83</string>
			</dict>
		</dict>
		<dict>
			<key>name</key>
			<string>JavaScript Functions</string>
			<key>scope</key>
			<string>variable.function.js</string>
			<key>settings</key>
			<dict>
				<key>foreground</key>
				<string>#b58900</string>
			</dict>
		</dict>
		<dict>
			<key>name</key>
			<string>JavaScript Classes</string>
			<key>scope</key>
			<string>variable.language.this.js, variable.language.super.js</string>
			<key>settings</key>
			<dict>
				<key>foreground</key>
				<string>#d33682</string>
			</dict>
		</dict>
```

**Comment by yeefom** [comment]
Just tried. It looks so good! Only noticed two minor issues: 1. all the operators are grey (I forgot to point this out in the first place!), 2. semicolons and commas are orange inside a class, but normal (grey) if they are not in a class. 

![](https://cldup.com/npJWkz9ebe.png)

**Comment by braver** [comment]
Yeah, operators are grey throughout the theme (or it *should* be...). I'm keeping that as is. However, the punctuation should also be grey. This corrects it:

```xml
		<dict>
			<key>name</key>
			<string>JavaScript punctation inside class</string>
			<key>scope</key>
			<string>meta.class punctuation</string>
			<key>settings</key>
			<dict>
				<key>foreground</key>
				<string>#657b83</string>
			</dict>
		</dict>
```

**Comment by yeefom** [comment]
Ok keeping operators color as it is sounds good. And the punctuations are also fixed. I’m so happy that Solarized is back as my theme again. Thank you so much? Are you going to commit it or just wait to see how things go?

**Comment by braver** [comment]
I just released this in 1.5.2! Thanks for your cooperation :)
